### PR TITLE
[DON'T MERGE] Feature/add payment expiry

### PR DESCRIPTION
**Pull request by erasmus**
Please double-check *everything*. Can only be deployed after the payment expiry feature is live --> https://github.com/satoshipay/user-stories/issues/256

### CODE REVIEW

File: source/includes/_api.md [hunk 1]
```diff
@@ -152,6 +152,7 @@ Property | Type      | Description
 `sharedSecret` | *string*  | Shared secret information which will be used to sign the `paymentReceipt` used to [authenticate](#retriving-auth) user during digital goods [retrieval](#retrieving-goods).
 `url`    | *string*  | URL of the web page which contains the good. Used as a reference in the [Dashboard](https://dashboard.satoshipay.io/performance/goods).
 `title`  | *string*  | Title of the good for reference in the provider dashboard.
+`purchaseValidityPeriod`  | *string*  | Time in milliseconds until the good's payment expires..
```
Comment (TorstenStueber) [inline source/includes/_api.md]: This is not quite right. We use the package `ms` to allow for every nice way to format this duration:

https://www.npmjs.com/package/ms

File: source/includes/_api.md [hunk 2]
```diff
@@ -289,6 +294,7 @@ Property | Type      | Description
 `sharedSecret` | *string*  | Shared secret information which will be used to sign the `paymentReceipt` used to [authenticate](#retriving-auth) user during digital goods [retrieval](#retrieving-goods).
 `url`    | *string*  | URL of the web page which contains the good. Used as a reference in the [Dashboard](https://dashboard.satoshipay.io/performance/goods).
 `title`  | *string*  | Title of the good for reference in the provider dashboard.
+`purchaseValidityPeriod`  | *string*  | Time in milliseconds until the good's payment expires.
```
Comment (TorstenStueber) [inline source/includes/_api.md]: Same here

File: source/includes/_api.md [hunk 3]
```diff
@@ -361,6 +370,7 @@ Property | Type      | Required | Description
 `sharedSecret` | *string*  | yes      | Shared secret information which will be used to create `paymentReceipt` used to [authenticate](#retriving-auth) user during digital goods [retrieval](#retrieving-goods).
 `url`    | *string*  | yes      | URL of the web page which contains the good. Used as a reference in the [Dashboard](https://dashboard.satoshipay.io/performance/goods).
 `title`  | *string*  | yes      | Title of the good for reference in the provider dashboard.
+`purchaseValidityPeriod`  | *string*  | no      | Time in milliseconds until the good's payment expires.
```
Comment (TorstenStueber) [inline source/includes/_api.md]: And here.

File: source/includes/_api.md [hunk 4]
```diff
@@ -429,6 +439,7 @@ Property | Type      | Required | Description
 `sharedSecret` | *string*  | no       | Shared secret information which will be used to create `paymentReceipt` used to [authenticate](#retriving-auth) user during digital goods [retrieval](#retrieving-goods).
 `url`    | *string*  | no       | URL of the web page which contains the good. Used as a reference in the [Dashboard](https://dashboard.satoshipay.io/performance/goods).
 `title`  | *string*  | no       | Title of the good for reference in the provider dashboard.
+`purchaseValidityPeriod`  | *string*  | no      | Time in milliseconds until the good's payment expires.
```
Comment (TorstenStueber) [inline source/includes/_api.md]: And here.

**Review by TorstenStueber — DISMISSED** [review]
I recommend to make use of the `ms` package and write, e.g., "1d" instead of "86400000" in the examples below.

File: source/includes/_api.md (hunk 1 above)
Comment (erasmus) [reply source/includes/_api.md]: please make a suggestion for how to phrase this in this documentation. What is acceptable input?

Comment (TorstenStueber) [reply source/includes/_api.md]: Duration until the good's payment expires, provided as a string in the form "<number> <unit>", where <unit> can be "minutes", "hours", "days", "weeks", "years" (or common abbreviations such as "min", "w", "y").

**Comment by erasmus** [comment]
Checks are failing. I don't see anything I can do to fix.

File: source/includes/_api.md [hunk 5]
```diff
@@ -173,7 +175,8 @@ curl https://api.satoshipay.io/v2/goods/ \
        "price": 200000000,
        "asset": "XLM",
        "title": "Nihil placeat sapiente ut eaque assumenda et reprehenderit quos ab.",
-       "url": "http://example.org/post1"
+       "url": "http://example.org/post1",
+       "purchaseValidityPeriod": "1m"
```
Comment (andywer) [inline source/includes/_api.md]: `"1m"` is not be the best sample, since users might think it means "1 month" (it's 1 minute).

Comment (erasmus) [reply source/includes/_api.md]: sure, but they get to see the response where it says 60000. Should be a valuable lesson for all ;)

**Review by andywer — APPROVED** [review]

**Pull request merged**